src/_selectors/AssetSelectors.js

```diff
@@ -3,6 +3,7 @@ import { toPlainJS } from '../_utils/ObjectUtils';
 
 export const assetsSelector = state => toPlainJS(state.assets);
 
+// return a tree structure, hierarchy as [market -> submarket -> symbol]
 export const marketTreeSelector = createSelector(
```

Comment:
BTW, I am not sure the way I implemented the market tree is ideal, or should have used arrays instead of objects. (just a note :)